### ISSUE TITLE
v0.1.1 Custom component compile fails with Syntax Error

### ISSUE THREAD

**Issue by jonsuh** [ChristopherBiscardi/gatsby-mdx#93]
**Describe the bug**
Compile fails via a `SyntaxError Expected corresponding JSX closing tag` after upgrading to `0.1.1` from `0.0.15`.

```
ERROR in ./src/pages/index.mdx
Module build failed (from ./node_modules/gatsby/dist/utils/babel-loader.js):
SyntaxError: /gatsby/src/pages/index.mdx: Expected corresponding JSX closing tag for <Linky> (16:0)

  14 | 
  15 | <MDXTag name="p" components={components}>{`Click `}<Linky href="https://github.com">{`here`}
> 16 | </MDXTag></MDXTag>
     | ^
  17 | 
...
```

<details>
  <summary>Verbose error</summary><p>

  ```
  ./src/pages/index.mdx
  Module build failed (from ./node_modules/gatsby/dist/utils/babel-loader.js):
  SyntaxError: /gatsby/src/pages/uses/index.mdx: Expected corresponding JSX closing tag for <Linky> (16:0)

    14 | 
    15 | <MDXTag name="p" components={components}>{`Click `}<Linky href="https://github.com">{`here`}
  > 16 | </MDXTag></MDXTag>
       | ^
    17 | 
    18 | export const _frontmatter = {};
    19 | 
      at _class.raise (/gatsby/node_modules/@babel/parser/lib/index.js:3893:15)
      at _class.jsxParseElementAt (/gatsby/node_modules/@babel/parser/lib/index.js:3559:18)
      at _class.jsxParseElementAt (/gatsby/node_modules/@babel/parser/lib/index.js:3532:34)
      at _class.jsxParseElementAt (/gatsby/node_modules/@babel/parser/lib/index.js:3532:34)
      at _class.jsxParseElement (/gatsby/node_modules/@babel/parser/lib/index.js:3585:19)
      at _class.parseExprAtom (/gatsby/node_modules/@babel/parser/lib/index.js:3592:21)
      at _class.parseExprSubscripts (/gatsby/node_modules/@babel/parser/lib/index.js:5898:21)
      at _class.parseMaybeUnary (/gatsby/node_modules/@babel/parser/lib/index.js:5877:21)
      at _class.parseExprOps (/gatsby/node_modules/@babel/parser/lib/index.js:5786:21)
      at _class.parseMaybeConditional (/gatsby/node_modules/@babel/parser/lib/index.js:5758:21)
      at _class.parseMaybeAssign (/gatsby/node_modules/@babel/parser/lib/index.js:5705:21)
      at _class.parseFunctionBody (/gatsby/node_modules/@babel/parser/lib/index.js:6887:24)
      at _class.parseArrowExpression (/gatsby/node_modules/@babel/parser/lib/index.js:6847:10)
      at _class.parseParenAndDistinguishExpression (/gatsby/node_modules/@babel/parser/lib/index.js:6463:12)
      at _class.parseExprAtom (/gatsby/node_modules/@babel/parser/lib/index.js:6254:21)
      at _class.parseExprAtom (/gatsby/node_modules/@babel/parser/lib/index.js:3594:52)
      at _class.parseExprSubscripts (/gatsby/node_modules/@babel/parser/lib/index.js:5898:21)
      at _class.parseMaybeUnary (/gatsby/node_modules/@babel/parser/lib/index.js:5877:21)
      at _class.parseExprOps (/gatsby/node_modules/@babel/parser/lib/index.js:5786:21)
      at _class.parseMaybeConditional (/gatsby/node_modules/@babel/parser/lib/index.js:5758:21)
      at _class.parseMaybeAssign (/gatsby/node_modules/@babel/parser/lib/index.js:5705:21)
      at _class.parseExportDefaultExpression (/gatsby/node_modules/@babel/parser/lib/index.js:8182:22)
   @ ./.cache/sync-requires.js 19:56-126
   @ ./.cache/app.js
   @ multi ./node_modules/react-hot-loader/patch.js (webpack)-hot-middleware/client.js?path=http://localhost:8000/__webpack_hmr&reload=true&overlay=false ./.cache/app
  ```
</p></details>

---

**To Reproduce**
Page source:
```
import { MDXProvider } from "@mdx-js/tag"

import Linky from "components/Linky"

export default ({ children }) => (
  <MDXProvider>
    {children}
  </MDXProvider>
)

Click <Linky href="https://github.com">here</Linky>
```

Component source:

```
import React from "react"

const Linky = ({ children, href = "" }) => (
  <a href={href}>
    {children}
  </a>
)

export default Linky
```

**Additional context**
Build is successful with version `0.0.15`

**Comment by ChristopherBiscardi** [comment]
Thanks for the issue. This is because of the new gatsby-remark-* plugin support that we added. To support the older plugins we needed to convert html to JSX and this is how the inline component is parsed:

```
{ type: 'paragraph',
  children:
   [ { type: 'text', value: 'An inline ', position: [Position] },
     { type: 'html',
       value: '<Link to="/somewhere">',
       position: [Position] },
     { type: 'text', value: 'react component', position: [Position] },
     { type: 'html', value: '</Link>', position: [Position] } ],
```

So the JSX detection we were using failed on just a closing tag: `</Link>`

**Comment by ChristopherBiscardi** [comment]
Released the fix in 0.1.2

**Comment by jonsuh** [comment]
@ChristopherBiscardi You’re awesome! Thank you for fixing this issue—I can confirm that `0.1.2` fixes this issue.

Furthermore, thank you so much for `gatsby-mdx`!
